DataFormats/Detectors/TRD/include/DataFormatsTRD/RawDataStats.h

```diff
@@ -30,136 +30,83 @@ namespace o2::trd
+  DigitParsingExitInWrongState,        // exiting parsing in the wrong state ... got to the end of the buffer in wrong state.
+  DigitHCHeaderMismatch,               // the half-chamber ID from the digit HC header is not consistent with the one expected from the link ID
+  TrackletHCHeaderFailure,             // either reserved bit not set or HCID is not what was expected from RDH
+  TrackletMCMHeaderSanityCheckFailure, // MCMHeader sanity check failure, LSB or MSB not set
+  TrackletDataWrongOrdering,           // the tracklet data is not arriving in increasing MCM order
+  TrackletDataDuplicateMCM,            // we see more than one TrackletMCMHeader for the same MCM
+  TrackletNoTrackletEndMarker,         // got to the end of the buffer with out finding a tracklet end marker.
+  TrackletExitingNoTrackletEndMarker,  // got to the end of the buffer exiting tracklet parsing with no tracklet end marker
+  DigitHeaderCountGT3,                 // digital half chamber header had more than 3 additional words expected by header. most likely corruption above somewhere.
+  DigitHeaderWrongType,                // expected digit header, but could not determine type
+  HalfCRUSumLength,                    // if the HalfCRU headers summed lengths wont fit into the buffer, implies corruption, its a faster check than the next one.
+  BadRDHMemSize,                       // RDH memory size is supposedly zero
+  BadRDHFEEID,                         // RDH parsing failure for reasons in the word
+  BadRDHEndPoint,                      // RDH parsing failure for reasons in the word
+  BadRDHOrbit,                         // RDH parsing failure for reasons in the word
+  BadRDHCRUID,                         // RDH parsing failure for reasons in the word
+  BadRDHPacketCounter,                 // RDH packet counter not incrementing
+  HalfCRUCorrupt,                      // if the HalfCRU headers has values out of range, corruption is assumed.
+  DigitHCHeader1Problem,               // multiple instances of Digit HC Header 1
+  DigitHCHeader2Problem,               // multiple instances of Digit HC Header 2
+  DigitHCHeader3Problem,               // multiple instances of Digit HC Header 3
+  DigitHCHeaderSVNMismatch,            // svn version information has changed in the DigitHCHeader3.
+  TrackletsReturnedMinusOne,           // trackletparsing returned -1, data was dumped;
+  FEEIDIsFFFF,                         // RDH is in error, the FEEID is 0xffff
+  FEEIDBadSector,                      // RDH is in error, the FEEID.supermodule is not a valid value.
+  HalfCRUBadBC,                        // the BC in the half-CRU header is so low that the BC shift would make it negative
+  TRDLastParsingError                  // This is to keep QC happy until we can change it there as well.
 };
 
-static std::unordered_map<int, std::string> ParsingErrorsString = {
+static const std::unordered_map<int, std::string> ParsingErrorsString = {
```

Comment:
Going through this a bit.
Isnt ParsingErrors::BadRDHMemSize missing here?
This would cause an exception if the rdh memory is invalid, which we probably do not want :smile: 
```cxx
  if (o2::raw::RDHUtils::getMemorySize(rdh) <= 0) {
    if (mMaxWarnPrinted > 0) {
      LOG(warn) << "Received RDH header with invalid memory size (<= 0) ";
      checkNoWarn();
    }
    incrementErrors(BadRDHMemSize, -1, fmt::format("Received RDH header with invalid memory size (<= 0) "));
    return false;
  }
}
```

Reply:
Thanks, indeed. I have added a runtime check on the size of the map and the number of entries in the enum now. A compile time check with static assert is unfortunately not possible with the map since it does not have a `constexpr` constructor